examples/incompressible-flow/2d-mms/MMS_QUAD/MMS_2D_steady.prm

```diff
@@ -0,0 +1,161 @@
+ set number = 4
+  subsection bc 0
+    set id   = 0
+    set type = function
+    subsection u
+      set Function expression = 0
+    end
+    subsection v
+      set Function expression = 0
+    end
+  end
+  subsection bc 1
+    set id   = 1
+    set type = function
+    subsection u
+      set Function expression = 0
+    end
+    subsection v
+      set Function expression = 0
+    end
+  end
+  subsection bc 2
+    set id   = 2
+    set type = function
+    subsection u
+      set Function expression = 0
+    end
+    subsection v
+      set Function expression = 0
+    end
+  end
+  subsection bc 3
+    set id   = 3
+    set type = function
+    subsection u
+      set Function expression = 0
+    end
+    subsection v
+      set Function expression = 0
+    end
+  end
+end
+
+# --------------------------------------------------
+# Source term
+#---------------------------------------------------
+subsection source term
+    subsection fluid dynamics
+      set enable                  = true
+      set Function constants  =  nu=1
+      set Function expression = pi*sin(pi*y)*(-16*pi*nu*sin(pi*x)*sin(pi*x)*cos(pi*y)+ 4*pi*nu*cos(pi*y)+4*sin(pi*x)*sin(pi*x)*sin(pi*x)*sin(pi*y)*cos(pi*x)+cos(pi*x)); pi*sin(pi*x)*(16*pi*nu*sin(pi*y)*sin(pi*y)*cos(pi*x)- 4*pi*nu*cos(pi*x)+4*sin(pi*x)*sin(pi*y)*sin(pi*y)*sin(pi*y)*cos(pi*y)+cos(pi*y)); 0
+    end
+end
+
+
+# --------------------------------------------------
+# Analytical Solution
+#---------------------------------------------------
+subsection analytical solution
+  set enable                 = true
+  # set filename               = L2Error_2D_incompressibleNS_steady
+    subsection uvwp
+        set Function expression = -2*sin(pi*x) * sin(pi*x) * cos(pi*y) * sin(pi*y) ; 2*cos(pi*x) * sin(pi*x) * sin(pi*y) * sin(pi*y); sin(pi*x)*sin(pi*y)
+    end
+end
+
+#---------------------------------------------------
+# FEM
+#---------------------------------------------------
+
+subsection FEM
+  set velocity order = {{Poly_deg_u}}
+  set pressure order = {{Poly_deg_p}}
+end
+
+# --------------------------------------------------
+# Mesh Adaptation Control
+#---------------------------------------------------
+subsection mesh adaptation
+  set type                    = uniform
+  #set max number elements     = 20000
+  #set max refinement level    = 6
+  #set min refinement level    = 6
+  #set frequency               = 1
+  #set fraction refinement     = 0.1
+  #set fraction coarsening     = 0.1
+end
+
+
+#---------------------------------------------------
+# Non-Linear Solver Control
+#---------------------------------------------------
+
+subsection non-linear solver
+  subsection fluid dynamics
+    set tolerance = 1e-10
+    set verbosity = verbose
+  end
+end
+
+#---------------------------------------------------
+# Linear Solver Control
+#---------------------------------------------------
+
+subsection linear solver
+  subsection fluid dynamics
+    set verbosity = verbose
+  end
```

Comment:
I suggest using an AMG preconditioner here. It will make things a lot simpler/faster in 2D